flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/utils/SqlScriptReader.java

```diff
@@ -0,0 +1,121 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.client.cli.utils;
+
+import javax.annotation.Nullable;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A utility to read and parse content of a SQL script. The SQL script is located in "resources/sql"
+ * path and in the "xx.q" file name pattern. The SQL script is executed and tested by {@link
+ * org.apache.flink.table.client.cli.CliClientITCase}.
+ */
+public final class SqlScriptReader implements AutoCloseable {
+
+    private final BufferedReader reader;
+    private String currentLine;
+
+    public static List<TestSqlStatement> parseSqlScript(String in) {
+        try (SqlScriptReader sqlReader = new SqlScriptReader(in)) {
+            return sqlReader.parseSqlScript();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private SqlScriptReader(String input) {
+        this.reader = new BufferedReader(new StringReader(input));
+    }
+
+    private List<TestSqlStatement> parseSqlScript() throws IOException {
+        List<TestSqlStatement> specs = new ArrayList<>();
+        TestSqlStatement spec;
+        while ((spec = readNext()) != null) {
+            specs.add(spec);
+        }
+        return specs;
+    }
+
+    private void readLine() throws IOException {
+        this.currentLine = reader.readLine();
+    }
+
+    private @Nullable TestSqlStatement readNext() throws IOException {
+        StringBuilder commentLines = new StringBuilder();
+        StringBuilder sqlLines = new StringBuilder();
+        ReadingStatus status = ReadingStatus.BEGINNING;
+        readLine();
+        while (currentLine != null) {
+            switch (status) {
+                case BEGINNING:
+                    if (currentLine.startsWith("#") || currentLine.trim().length() == 0) {
+                        commentLines.append(currentLine).append("\n");
+                        // continue reading if not reach SQL statement
+                        readLine();
+                    } else {
+                        // if current currentLine is not comment and empty currentLine, begin to
+                        // read SQL
+                        status = ReadingStatus.SQL_STATEMENT;
+                    }
+                    break;
+
+                case SQL_STATEMENT:
+                    sqlLines.append(currentLine).append("\n");
+                    if (currentLine.trim().endsWith(";")) {
```

Comment:
how about multiple sql statements (e.g. insert statement)

Reply:
Do you mean `STATEMENT SET`? That should be supported naturally, because the STATEMENT SET will be separated into several statement, and each one is a `SQL_STATEMENT` and will have a result: either `!ok`, or `!info`, `!error`.